server/base/src/main/java/org/apache/accumulo/server/ServerContext.java

```diff
@@ -182,6 +191,10 @@ public SiteConfiguration getSiteConfiguration() {
 
   @Override
   public AccumuloConfiguration getConfiguration() {
+    return serverConfFactory.get().getResourceGroupConfiguration();
```

Comment:
Is there ever a situation where this wouldn't apply? Or is everything using ServerContext using the default resource group?

Also one note about the default resource group... it seems to be a convention, but not necessarily strictly protected (e.g. can you delete the default resource group?)

Reply:
> Is there ever a situation where this wouldn't apply? Or is everything using ServerContext using the default resource group?

The ServerConfigurationFactory is created using the ResourceGroupId of the server process. It should return the ResourceGroupConfiguration for the configured resource group, it should not always be default.

> Also one note about the default resource group... it seems to be a convention, but not necessarily strictly protected (e.g. can you delete the default resource group?)

`ManagerClientServiceHandler.removeResourceGroupNode` throws an exception if the default resource group is passed as an argument.

Reply:
My original implementation did what you are describing. @keith-turner suggested that we should separate them and provide access to each, discussion at https://github.com/apache/accumulo/pull/5749#discussion_r2231627609. You can see how I had the code originally looking at the ServerConfigurationFactory class in https://github.com/apache/accumulo/commit/febdd079e8235d9f83058c3db09b349e15779517.